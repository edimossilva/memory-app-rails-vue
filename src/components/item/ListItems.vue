<template>
  <div>
    <h1>ListItems</h1>
    <div v-for="item in items" :key="item.key">
      key: {{ item.key }}, value: {{ item.value }}
    </div>
  </div>
</template>

<script>
import { mapState } from "vuex";

export default {
  name: "ListItems",
  data() {
    return {
      item: { key: "", value: "" },
    };
  },
  computed: {
    ...mapState(["items"]),
  },
};
</script>
