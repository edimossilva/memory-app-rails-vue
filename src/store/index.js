import Vue from "vue";
import Vuex from "vuex";

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    items: [],
  },
  mutations: {
    addItem(state, item) {
      state.items.push(item);
    },
    // mutate(state, payload) {
    //   state[payload.property] = payload.value;
    // },
  },
  actions: {
    addItem({ commit }, item) {
      commit("addItem", { ...item });
    },
  },
  modules: {},
});
