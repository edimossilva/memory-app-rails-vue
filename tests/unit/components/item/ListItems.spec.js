import { shallowMount } from "@vue/test-utils";
import ListItems from "@/components/item/ListItems.vue";
import Vue from "vue";
import VueRouter from "vue-router";
import store from "@/store/index";

Vue.use(VueRouter);

const mockItems = [
  { key: "mockKey1", value: "mockValue1" },
  { key: "mockKey2", value: "mockValue2" },
];
describe("ListItems.vue", () => {
  it("Matchs Snapshot", () => {
    const wrapper = shallowMount(ListItems, {
      store,
      computed: { items: () => mockItems },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
