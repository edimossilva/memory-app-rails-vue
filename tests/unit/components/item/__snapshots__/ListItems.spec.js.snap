// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItems.vue Matchs Snapshot 1`] = `
<div>
  <h1>ListItems</h1>
  <div>
    key: mockKey1, value: mockValue1
  </div>
  <div>
    key: mockKey2, value: mockValue2
  </div>
</div>
`;
